### PR TITLE
feat: warn when state sync settings are set while `enable = false`

### PR DESCRIPTION
**Pull request by najeal**
## Description

Closes https://github.com/celestiaorg/celestia-app/issues/3736

This adds warn logs when state sync settings are set while `enable = false` as it may lead to unexpected behavior whereas Config is valid.

#### PR checklist

- [X] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use
  [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: libs/log/tm_logger.go [hunk 1]
```diff
@@ -79,6 +81,16 @@ func (l *tmLogger) Error(msg string, keyvals ...interface{}) {
 	}
 }
 
+// Warnlogs a message at warning Error.
```
Comment (rootulp) [inline libs/log/tm_logger.go:84]:
```suggestion
// Warn logs a message at level Warn.
```

File: libs/log/tm_logger_test.go [hunk 2]
```diff
@@ -66,6 +66,29 @@ func TestDebug(t *testing.T) {
 	}
 }
 
+func TestWarn(t *testing.T) {
+	var bufWarn bytes.Buffer
+
+	ld := log.NewTMLogger(&bufWarn)
+	ld.Debug("Client initialized with old header (trusted is more recent)",
```
Comment (rootulp) [inline libs/log/tm_logger_test.go:73]: [blocking] how does this test Warn if it's using `Debug`? Should this be `Warn`?

```suggestion
	ld.Warn("Client initialized with old header (trusted is more recent)",
```

**Review by rootulp — COMMENTED** [review]
Great work overall! I have a few nits and one possible change: it seems odd to issue a warning for the default state sync config. Should we modify the behavior here to only warn if a user has specified a non-empty RPC servers?

Also, this PR could be upstreamed to CometBFT after we merge it here.

**Comment by rootulp** [comment]
BTW updated PR title to conform to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)

**Comment by najeal** [comment]
> Also, this PR could be upstreamed to CometBFT after we merge it here.

I would be glad to open a PR to CometBFT when this one is merged.

> BTW updated PR title to conform to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)

Sorry about that

I committed changes to only return possible misconfigurations when RPCServers are specified

**Review by evan-forbes — APPROVED** [review]
yo nice @najeal thanks for this!

**Review by rach-id — COMMENTED** [review]
LGTM :+1:

**Review by rach-id — APPROVED** [review]

**Comment by najeal** [comment]
Shall I add something for the changelog ? I saw some indications about Changelog in the CONTRIBUTING.md but I don't see any change in .changelog since last year.

**Comment by rootulp** [comment]
No need to add a changelog because we don't use the changelog in this repo. I think it exists because upstream uses it.

**Review by rootulp — APPROVED** [review]

**Pull request merged**